app/src/main/java/org/fossasia/susi/ai/adapters/viewholders/LinkPreviewViewHolder.java

```diff
@@ -43,7 +72,179 @@
+        if (model.getWebLinkData() == null) {
+            LinkPreviewCallback linkPreviewCallback = new LinkPreviewCallback() {
+                @Override
+                public void onPre() {
+                    previewImageView.setVisibility(View.GONE);
+                    descriptionTextView.setVisibility(View.GONE);
+                    titleTextView.setVisibility(View.GONE);
+                    previewLayout.setVisibility(View.GONE);
+                }
+
+                @Override
+                public void onPos(final SourceContent sourceContent, boolean b) {
+                    realm.beginTransaction();
+                    Realm realm = Realm.getDefaultInstance();
+                    WebLink link = realm.createObject(WebLink.class);
+
+                    if(sourceContent != null) {
+
+                        if (!sourceContent.getDescription().isEmpty()) {
+                            Log.d(TAG, "onPos: " + sourceContent.getDescription());
+                            previewLayout.setVisibility(View.VISIBLE);
+                            descriptionTextView.setVisibility(View.VISIBLE);
+                            descriptionTextView.setText(sourceContent.getDescription());
+                        }
+
+                        if (!sourceContent.getTitle().isEmpty()) {
+                            Log.d(TAG, "onPos: " + sourceContent.getTitle());
+                            previewLayout.setVisibility(View.VISIBLE);
+                            titleTextView.setVisibility(View.VISIBLE);
+                            titleTextView.setText(sourceContent.getTitle());
+                        }
+
+                        link.setBody(sourceContent.getDescription());
+                        link.setHeadline(sourceContent.getTitle());
+                        link.setUrl(sourceContent.getUrl());
+                        url = sourceContent.getFinalUrl();
+
+                        final List<String> imageList = sourceContent.getImages();
+
+                        if (imageList == null || imageList.size() == 0) {
+                            previewImageView.setVisibility(View.GONE);
+                            link.setImageURL("");
+                        } else {
+                            previewImageView.setVisibility(View.VISIBLE);
+                            Picasso.with(currContext).load(imageList.get(0))
+                                    .fit().centerCrop()
+                                    .into(previewImageView);
+                            link.setImageURL(imageList.get(0));
+                        }
+                    }
+
+                    model.setWebLinkData(link);
+                    realm.copyToRealmOrUpdate(model);
+                    realm.commitTransaction();
+                }
+            };
+
+            if (model != null) {
+                List<String> urlList = ((MainActivity)currContext).recyclerAdapter.extractLinks(model.getContent());
```

Comment:
extractLinks is static. Use this ChatFeedRecyclerAdapter.extractLinks(model.getContent());